obsreport/obsreport_test.go

```diff
@@ -219,62 +219,60 @@ func TestReceiveMetricsOp(t *testing.T) {
 }
 
 func TestScrapeMetricsDataOp(t *testing.T) {
-	tt, err := obsreporttest.SetupTelemetry()
-	require.NoError(t, err)
-	t.Cleanup(func() { require.NoError(t, tt.Shutdown(context.Background())) })
-
-	parentCtx, parentSpan := tt.TracerProvider.Tracer("test").Start(context.Background(), t.Name())
-	defer parentSpan.End()
+	testTelemetry(t, func(tt obsreporttest.TestTelemetry, registry *featuregate.Registry) {
```

Comment:
This API is horrible, makes everything be super indented...

/cc @paivagustavo

Reply:
@jmacd suggested the creation of new top level function tests as https://github.com/open-telemetry/opentelemetry-collector/pull/6423/commits/5961f605ec41cab61ac1abd1f75d6f3a3cc06657, which keeps the same indentation for the existing tests and can be easily removed once we're done with the transition.

cc @moh-osman3

Reply:
Let me know if my latest commit matches what was suggested.